syntax = "proto3";

package minos;

service Minos {
    rpc GetPasswdAll(Empty) returns (PasswdList);
    rpc GetPasswdByName(ByName) returns (Passwd);
    rpc GetPasswdByUid(ByID) returns (Passwd);
    rpc GetGroupAll(Empty) returns (GroupList);
    rpc GetGroupByName(ByName) returns (Group);
    rpc GetGroupByGid(ByID) returns (Group);
    rpc GetShadowAll(Empty) returns (ShadowList);
    rpc GetShadowByName(ByName) returns (Shadow);
}

message Empty {}

message ByName {
    string name = 1;
}

message ByID {
    uint32 id = 1;
}

message Passwd {
    string pw_name = 1;
    string pw_passwd = 2;
    uint32 pw_uid = 3;
    uint32 pw_gid = 4;
    string pw_gecos = 5;
    string pw_dir = 6;
    string pw_shell = 7;
}

message Group {
    string gr_name = 1;
    string gr_passwd = 2;
    uint32 gr_gid = 3;
    repeated string gr_mem = 4;
}

message Shadow {
    string sp_namp = 1;
    string sp_pwdp = 2;
    int32 sp_lstchg = 3;
    int32 sp_min = 4;
    int32 sp_max = 5;
    int32 sp_warn = 6;
    int32 sp_inact = 7;
    int32 sp_expire = 8;
    uint32 sp_flag = 9;
}

message PasswdList {
    repeated Passwd entries = 1;
}

message GroupList {
    repeated Group entries = 1;
}

message ShadowList {
    repeated Shadow entries = 1;
}
